cl/tests/unit/test_free_look_email_handler.py

```diff
@@ -0,0 +1,92 @@
+import json
+import pytest
+import requests
+import requests_mock
+from free_look_email import app
+
+@pytest.fixture()
+def spam_failure_ses_event():
+    with open('./events/ses-spam-failure.json') as f:
+        data = json.load(f)
+    return data
+
+def test_spam_failure(spam_failure_ses_event):
+    response = app.handler(spam_failure_ses_event, '')
+    data = json.loads(response['body'])
+
+    assert response['statusCode'] == 424
+    assert 'spam_verdict' in data
+    assert data['spam_verdict']['status'] == 'FAILED'
+
+@pytest.fixture()
+def virus_failure_ses_event():
+    with open('./events/ses-virus-failure.json') as f:
+        data = json.load(f)
+    return data
+
+def test_virus_failure(virus_failure_ses_event):
+    response = app.handler(virus_failure_ses_event, '')
+    data = json.loads(response['body'])
+
+    assert response['statusCode'] == 424
+    assert 'virus_verdict' in data
+    assert data['virus_verdict']['status'] == 'FAILED'
+
+@pytest.fixture()
+def spf_failure_ses_event():
+    with open('./events/ses-spf-failure.json') as f:
+        data = json.load(f)
+    return data
+
+def test_spf_failure(spf_failure_ses_event):
+    response = app.handler(spf_failure_ses_event, '')
+    data = json.loads(response['body'])
+
+    assert response['statusCode'] == 424
+    assert 'spf_verdict' in data
+    assert data['spf_verdict']['status'] == 'FAILED'
+
+@pytest.fixture()
+def dkim_failure_ses_event():
+    with open('./events/ses-dkim-failure.json') as f:
+        data = json.load(f)
+    return data
+
+def test_dkim_failure(dkim_failure_ses_event):
+    response = app.handler(dkim_failure_ses_event, '')
+    data = json.loads(response['body'])
+
+    assert response['statusCode'] == 424
+    assert 'dkim_verdict' in data
+    assert data['dkim_verdict']['status'] == 'FAILED'
+
+@pytest.fixture()
+def dmarc_failure_ses_event():
+    with open('./events/ses-dmarc-failure.json') as f:
+        data = json.load(f)
+    return data
+
+def test_dmarc_failure(dmarc_failure_ses_event):
+    response = app.handler(dmarc_failure_ses_event, '')
+    data = json.loads(response['body'])
+
+    assert response['statusCode'] == 424
+    assert 'dmarc_verdict' in data
+    assert data['dmarc_verdict']['status'] == 'FAILED'
+
+@pytest.fixture()
+def ses_event():
+    with open('./events/ses.json') as f:
+        data = json.load(f)
+    return data
+
+def test_success(ses_event, requests_mock):
+    requests_mock.post('http://host.docker.internal:8000/api/rest/v3/free-look-email/', json=json.dumps({'mail': {}, 'receipt': {}}))
```

Comment:
I'm confused what this line does? It doesn't return anything....I'm not familiar with `requests_mock`, but it looks like it doesn't _do_ anything either?

Reply:
This mocks out the requests service, so it ensures that this unit test is not actually making the service request. Instead, it is just saying "When requests receives a post to this address, return the json name param.